### PR TITLE
Example for __call__

### PR DESCRIPTION
**Pull request by edsaac**
Creates an Appliance object that can be either on or off. 

Each time the object is called as a function, it's state switches and a message is returned.  
Calling just the object will go through the `__repr__` dunder

### CODE REVIEW

**Review by trag1c — CHANGES_REQUESTED** [review]
I don't see why `switch` is its own method if it's only used in `__call__`

File: src/middle-earth/call.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    print(f"{tv =:}")      # __repr__
+    print(f"{tv() =:} ")   # __call__
+    print(f"{tv() =:} ")   # __call__
+    print(f"{tv() =:} ")   # __call__
+    print(f"{bool(tv) =:}")  # __bool__
+
+    print("\n*** PC ***")
+    print(f"{pc =:}")      # __repr__
+    print(f"{pc() =:} ")   # __call__
+    print(f"{bool(pc) =:}")  # __bool__
```
[the remark this answers is not in the document]
Comment (edsaac) [reply src/middle-earth/call.py:62]: Without the colon, the escape characters in the returned string are not escaped. 

|Case|Command|Output|
|:---|:----|:-----|
|Without `:` | `print{f"{tv() =}")`  | `tv() ='\tThe room TV is ON\t(from __call__)'` |
|With `:`       | `print{f"{tv() =:})"` | `tv() =  The room TV is ON       (from __call__) `|

[the remark this answers is not in the document]
Comment (edsaac) [reply src/middle-earth/call.py:62]: Maybe that could be an example for `__format__`

[the remark this answers is not in the document]
Comment (edsaac) [reply src/middle-earth/call.py]: Maybe that could be an example for `__format__`  ?

[the remark this answers is not in the document]
Comment (trag1c) [reply src/middle-earth/call.py]: TIL! I always used `f"{x = !s}"` for that:
```pycon
>>> x = "hello\tthere"
>>> print(f"{x = }")
x = 'hello\tthere'
>>> print(f"{x = !s}")
x = hello	there
```

[the remark this answers is not in the document]
Comment (trag1c) [reply src/middle-earth/call.py]: > Maybe that could be an example for `__format__`

Maybe maybe

**Comment by Lunarmagpie** [comment]
I think a better example for call is mimicking `functools.partial`. People probably wonder how that works as well.

File: src/middle-earth/call.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        message = "ON" if self.state else "OFF"
+        return f"\tThe {self.name} is {message}"  
+
+    def __call__(self) -> str:
+        """Switch the state of the Appliance and return it's status"""
+        self.state = not self.state
+        return self.status() + "\t(from __call__)"
+    
+    def __repr__(self) -> str:
+        return self.status() + "\t(from __repr__)"
```
Comment (MithicSpirit) [inline src/middle-earth/call.py:33]: I think that an fstring would be better than string concatenation. Also not too sure on the use of tabs here. Please discuss.

**Review by MithicSpirit — CHANGES_REQUESTED** [review]
Please format with black.

File: src/middle-earth/call.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    print(f"{tv =:}")      # __repr__
+    print(f"{tv() =:} ")   # __call__
+    print(f"{tv() =:} ")   # __call__
+    print(f"{tv() =:} ")   # __call__
+    print(f"{bool(tv) = }")  # __bool__
+
+    print("\n*** PC ***")
+    print(f"{pc =:}")      # __repr__
+    print(f"{pc() =:} ")   # __call__
+    print(f"{bool(pc) = }")  # __bool__
```
Comment (CircuitSacul) [inline src/middle-earth/call.py:54]: This is always True since you haven't implemented `__bool__`

**Review by CircuitSacul — CHANGES_REQUESTED** [review]

File: src/middle-earth/call.py (hunk 3 above)
Comment (edsaac) [reply src/middle-earth/call.py:54]: But `__bool__` is already implemented :thinking:

Comment (CircuitSacul) [reply src/middle-earth/call.py:54]: Yes, the default behavior is to return True. I assume it was supposed to return True if the appliance is on and False if the appliance is off, in which case you would need to override it?

Comment (CircuitSacul) [reply src/middle-earth/call.py:54]: Nevermind I'm blind

File: src/middle-earth/call.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+class Appliance:
+    """
+    An Appliance object can be either on and off.
+
+    A call to the object will result in switching the state
+    of the Appliance object.
+
+    """
+
+    def __init__(self, name: str, is_on: bool = False) -> None:
```
Comment (trag1c) [inline src/middle-earth/call.py:10]:
```suggestion
    def __init__(self, name: str, *, is_on: bool = False) -> None:
```

**Review by trag1c — CHANGES_REQUESTED** [review]

**Comment by edsaac** [comment]
The `functools.partial` idea sounds better than this dummy example. I'm closing this PR so `__call__`  can be returned to  the TO DO list on the tracking spreadsheet.

**Pull request closed without merging**